src/plugins/charts/public/services/theme/README.md

```diff
@@ -0,0 +1,92 @@
+
+> The current theme (light or dark) of Kibana is typically taken into account for the functions below.
+
+## `chartsDefaultBaseTheme`
+
+Default `baseTheme` from `@elastic/charts` (i.e. light).
+
+## `chartsDefaultTheme`
+
+Default `theme` from `@elastic/eui` (i.e. light).
+
+## `useChartsTheme` and `useChartsBaseTheme`
+
+A **React hook** for simple fetching of the correct EUI `theme` and `baseTheme`.
+
+```js
+import { npStart } from 'ui/new_platform';
+import { Chart, Settings } from '@elastic/charts';
+
+export const YourComponent = () => (
+  <Chart>
+    <Settings
+      theme={npStart.plugins.charts.theme.useChartsTheme()}
+      baseTheme={npStart.plugins.charts.theme.useChartsBaseTheme()}
+    />
+    {/* ... */}
+  </Chart>
+);
+```
+
+## `chartsTheme$` and `chartsBaseTheme$`
+
+An **`Observable`** of the current charts `theme` and `baseTheme`. Use this implementation for more flexible updates to the chart theme without full page refreshes.
+
+```tsx
+import { npStart } from 'ui/new_platform';
+import { EuiChartThemeType } from '@elastic/eui/src/themes/charts/themes';
+import { Subscription, combineLatest } from 'rxjs';
+import { Chart, Settings, Theme } from '@elastic/charts';
+
+interface YourComponentProps {};
+
+interface YourComponentState {
+  chartsTheme: EuiChartThemeType['theme'];
+  chartsBaseTheme: Theme;
+}
+
+export class YourComponent extends Component<YourComponentProps, YourComponentState> {
+  private subscriptions: Subscription[] = [];
+
+  public state = {
+    chartsTheme: npStart.plugins.charts.theme.chartsDefaultTheme,
+    chartsBaseTheme: npStart.plugins.charts.theme.chartsDefaultBaseTheme,
+  };
+
+  componentDidMount() {
+    this.subscription = combineLatest(
+      npStart.plugins.charts.theme.chartsTheme$,
+      npStart.plugins.charts.theme.chartsBaseTheme$
+    ).subscribe(([chartsTheme, chartsBaseTheme]) =>
+      this.setState({ chartsTheme, chartsBaseTheme })
+    );
+  }
+
+  componentWillUnmount() {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
+  }
+
+  public render() {
+    const { chartsBaseTheme, chartsTheme } = this.state;
+
+    return (
+      <Chart>
+        <Settings
+          theme={chartsTheme}
+          baseTheme={chartsBaseTheme}
+        />
+        {/* ... */}
+      </Chart>
+    );
+  }
+}
+```
+
+## Why have `theme` and `baseTheme`?
+
+The `theme` prop is a recursive partial `Theme` that overrides properties from the `baseTheme`. This allows changes to the `Theme` type in `@elastic/charts` without having to update the `@elastic/eui` themes or every `<Chart />`.
```

Comment:
```suggestion
The `theme` prop is a recursive partial `Theme` that overrides properties from the `baseTheme`. This allows changes to the `Theme` TS type in `@elastic/charts` without having to update the `@elastic/eui` themes for every `<Chart />`.
```

Reply:
Does this mean that every `<Chart />` instance now **must** provide both the `baseTheme` and the EUI provided `theme`?

Reply:
It's not a must. But since we added the background color to the theme for the contrast ratio logic, the background color defaults the theme background to a non-`transparent` colors respective of the mode (i.e. light and dark). So in order for the chart to work in both dark and light modes without a `baseTheme`, one would need to set the `theme.background.color` to `'transparent'`. This would avoid the contract logic though.

I'm not sure if `'transparent'` should just be the default for the background color or if you just want to add this to the EUI theme.

But on a more general note, I think it is helpful to have a `baseTheme` used in all elastic/charts `Charts` in kibana for times when we change the `Theme` object. This allows for nothing to break while in the meantime we update any changes to the EUI charts partial themes.

I raised the question in an EC issue, to see how we could do this better (see https://github.com/elastic/elastic-charts/issues/718). As of now, I think a HOC to handle this logic and allow overrides would be the most practicable option. Something like...

```tsx
const MyComponent = () => (
  <Chart>
    <ThemedSettings theme={EuiThemeOverrides} {...otherSettingsProps} />
  />
)
```

Reply:
Fixed spelling errors here 189843c